use null in sls file to map None object  GSoC

gqlo opened the pull request:
### What does this PR do?
Passing `None` with boot parameters in sls file is handled as a string. We could pass `null` so that it will be translated to Python `None` object by YAML. It makes more sense to do so.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltstack.com/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

Commit:
use null in sls file to map None object

add sls file example

reword doc

cbosdo commented on salt/modules/virt.py:
```diff
@@ -2265,9 +2265,9 @@ def update(
 
         To update any boot parameters, specify the new path for each. To remove any boot parameters,
-        pass a None object, for instance: 'kernel': ``None``.
+        pass null object, for instance: 'kernel': ``null``.
```
Don't mention `null` in that documentation since this is a YAML/SLS-specific value for `None`

cbosdo commented on salt/modules/virt.py:
```diff
@@ -2265,9 +2265,9 @@ def update(
+        pass null object, for instance: 'kernel': ``null``.
 
-        To switch back to BIOS boot, specify ('loader': ``None`` and 'nvram': ``None``)  or 'efi': ``False``
+        To switch back to BIOS boot, specify ('loader': ``null`` and 'nvram': ``null``)  or 'efi': ``False``
```
Same here

cbosdo reviewed (changes requested):
LGTM, but the doc changes are not needed. What could be done in the doc is to provide an example of SLS file with a `null` however.

cbosdo reviewed (dismissed):
:+1:

gqlo commented:
@dwoz

gqlo commented:
re-run all

gqlo commented:
re-run all

gqlo commented:
re-run full all

gqlo commented:
re-run full [ windows | all | etc. ]

gqlo commented:
re-run ci/pre-commit

gqlo commented:
re-run all

dwoz commented:
@gqlo Looks like black needs to run before the pre-commit check will pass

gqlo commented:
re-run all

cbosdo committed:
pylint: allow a few more 3rd party modules

Strangely the following errors keep blocking the pre-commit lint hook on virt
module while those never appear in the CI.

[W8410(3rd-party-module-not-gated), ] 3rd-party module import is not gated in a try/except: 'base64'
[W8410(3rd-party-module-not-gated), ] 3rd-party module import is not gated in a try/except: 'copy'
[W8410(3rd-party-module-not-gated), ] 3rd-party module import is not gated in a try/except: 'os'
[W8410(3rd-party-module-not-gated), ] 3rd-party module import is not gated in a try/except: 're'
[W8410(3rd-party-module-not-gated), ] 3rd-party module import is not gated in a try/except: 'shutil'

Add these modules to the allowed 3rd party modules to clear the errors.

cbosdo committed:
Update virt module and states and their tests to python3
